internal/eventstream/eventstream.go

```diff
@@ -0,0 +1,70 @@
+package eventstream
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/alecthomas/types/pubsub"
+)
+
+// EventStream is a stream of events that can be published and subscribed to, that update a materialized view
```

Comment:
This is not really accurate - the events themselves update the materialised view. TBH I don't love that too much as it seems like it violates encapsulation of the View...but it seems okay for now.